generator/views/ping_view.py

```diff
@@ -93,37 +93,68 @@ def to_lookml(self, bq_client) -> List[dict]:
+
     def get_measures(self, dimensions: List[dict], table: str) -> List[Dict[str, str]]:
         """Generate measures from a list of dimensions.
 
         When no dimension-specific measures are found, return a single "count" measure.
 
         Raise ClickException if dimensions result in duplicate measures.
         """
-        measures = {}
-
+        # Iterate through each of the dimensions and accumulate any measures
+        # that we want to include in the view. We pull out the client id first
+        # since we'll use it to calculate per-measure client counts.
+        client_id_field = self._get_client_id(dimensions, table)
+        measures = [
+            {
+                "name": "clients",
+                "type": "count_distinct",
+                "sql": f"${{{client_id_field}}}",
+            }
+        ]
         for dimension in dimensions:
             dimension_name = dimension["name"]
-            if dimension_name in {"client_id", "client_info__client_id"}:
-                measure = {
-                    "name": "clients",
-                    "type": "count_distinct",
-                    "sql": f"${{{dimension_name}}}",
-                }
-            elif dimension_name == "document_id":
-                measure = {"name": "ping_count", "type": "count"}
-            else:
-                continue
-            name = measure["name"]
-            if name in measures:
-                raise click.ClickException(
-                    f"duplicate measure {name!r} for table {table!r}"
-                )
-            measures[name] = measure
-
-        if len(measures) == 0:
+            if dimension_name == "document_id":
+                measures += [{"name": "ping_count", "type": "count"}]
+            elif "metrics__counter__" in dimension_name:
+                # handle the counters in the metric ping
+                name = "__".join(dimension_name.split("__")[1:])
+                measures += [
+                    {
+                        "name": name,
+                        "type": "sum",
+                        "sql": f"${{{dimension_name}}}",
+                    },
+                    {
+                        "name": f"{name}_client_count",
+                        "type": "count_distinct",
+                        "sql": (
+                            f"case when ${{{dimension_name}}} > 0 then "
+                            f"${{{client_id_field}}}"
+                        ),
+                    },
+                ]
+
+        # check if there are any duplicate values, and report the first one that
+        # shows up
+        names = [measure["name"] for measure in measures]
+        duplicates = [k for k, v in Counter(names).items() if v > 1]
+        if duplicates:
+            name = duplicates[0]
             raise click.ClickException(
-                f"Missing client_id and doc_id dimensions in {table!r}"
+                f"duplicate measure {name!r} for table {table!r}"
```

Comment:
I can't actually think of any situations where this would occur, outside of poorly specifying the loop above (e.g. only using `dimension_name[-1]`).